src/StructTypes.jl

```diff
@@ -9,8 +9,10 @@ StructType(x::T) where {T} = StructType(T)
 
 "Default `StructTypes.StructType` for types that don't have a `StructType` defined; this ensures objects must have an explicit `StructType` to avoid unanticipated issues"
 struct NoStructType <: StructType end
+struct SingletonType <: StructType end
 
-StructType(::Type{T}) where {T} = NoStructType()
+StructType(::Type{T}) where {T<:Exception} = NoStructType()
```

Comment:
I think it would be fine to make this `Struct()`; `Exception`s are usually just a couple of string/code fields, so it makes sense to treat them as "data types"

Reply:
removed, that's reasonable